packages/wrangler/src/config/validation.ts

```diff
@@ -652,33 +680,37 @@ function normalizeAndValidateAliases(
 }
 
 /**
- * Validate the `assets` configuration and return normalized values.
+ * Validate the `legacy_assets` configuration and return normalized values.
  */
-function normalizeAndValidateAssets(
+function normalizeAndValidateLegacyAssets(
 	diagnostics: Diagnostics,
 	configPath: string | undefined,
 	rawConfig: RawConfig
-): Config["assets"] {
+): Config["legacy_assets"] {
+	const mergedAssetsConfig = rawConfig["legacy_assets"] ?? rawConfig["assets"];
```

Comment:
nit: maybe rename this to `rawAssetsConfig`? `merged` makes me think the two configs were somehow merged into one, and might be misleading for others as well

Reply:
is being merged, added comment for clarity now!